syntax = "proto3";
package cplex_vrp;

message Matrix {
  repeated float time = 2 [packed=true];
  repeated float distance = 3 [packed=true];
  repeated float value = 4 [packed=true];
}

message TimeWindow {
  int64 start = 2;
  int64 end = 3;
}

message Service {
  repeated TimeWindow time_windows = 1;
  repeated float quantities = 2;
  uint32 matrix_index = 3;
  string id = 4;
  int64 duration = 5;
}

message Capacity {
  int32 limit = 1;
  float overload_multiplier = 2;
  bool counting = 3;
}

message Vehicle {
  string id = 1;
  repeated Capacity capacities = 2;
  TimeWindow time_window = 3;
  uint32 matrix_index = 4;
  int32 start_index = 5;
  int32 end_index = 6;
  int32 day_index = 7;
  uint32 value_matrix_index = 8;
}

message Route {
  string vehicle_id = 1;
  repeated string service_ids = 2;
}

message Problem {
  repeated Vehicle vehicles = 1;
  repeated Service services = 2;
  repeated Matrix matrices = 3;
  repeated Route routes = 4;
}
